### PR TITLE
Remove Python dependencies from Java CI.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR removes some Python dependencies from the Java CI scripts. It does not appear that these dependencies are needed. As we prepare to migrate this job to GitHub Actions in #12002, this cleanup will help us validate that we aren't dropping any real testing dependencies.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
rerun tests

**Review by jlowe — APPROVED** [review]

File: ci/gpu/java.sh [hunk 1]
```diff
@@ -95,37 +89,18 @@ conda activate rapids
...
-}
-
 ################################################################################
 # INSTALL - Install libcudf artifacts
 ################################################################################
 
 gpuci_logger "Installing libcudf & libcudf_kafka"
 gpuci_mamba_retry install -c ${CONDA_ARTIFACT_PATH} libcudf libcudf_kafka
 
-install_dask
```
Comment (ttnghia) [inline ci/gpu/java.sh:127]: Comment not about this line but the line 124 above: 
```
gpuci_logger "Installing libcudf & libcudf_kafka"
```
Should ` & libcudf_kafka` also be removed?

Comment (bdice) [reply ci/gpu/java.sh:127]: Good catch. I fixed this in the GitHub Actions PR already. Will fix.

**Review by ttnghia — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**